src/networking.c

```diff
@@ -1138,6 +1144,13 @@ void addReplyBulkSds(client *c, sds s) {
     _addReplyToBufferOrList(c, "\r\n", 2);
 }
 
+void addWritePreparedReplyBulkSds(writePreparedClient *c, sds s) {
+    _addReplyLongLongWithPrefix(c, sdslen(s), '$');
+    _addReplyToBufferOrList(c, s, sdslen(s));
+    sdsfree(s);
+    _addReplyToBufferOrList(c, "\r\n", 2);
+}
```

Comment:
I just realized while reading this code we aren't actually getting the type protections we thought we were getting. Since we need to do the cast from writePreparedClient to normal client. 

This code should look something like this:
```suggestion
void addWritePreparedReplyBulkSds(writePreparedClient *wpc, sds s) {
    client *c = (client *)wpc;
    _addReplyLongLongWithPrefix(c, sdslen(s), '$');
    _addReplyToBufferOrList(c, s, sdslen(s));
    sdsfree(s);
    _addReplyToBufferOrList(c, "\r\n", 2);
}
```

However, this isn't required because the definition of writePreparedClient should have been:
```
typedef struct writePreparedClient writePreparedClient;
```

Which is just an opaque pointer. Would you mind updating it and making the minor changes? Sorry to ask you to do it.

Reply:
okay, make sense to me. applied!